apiserver/facades/client/charms/client.go

```diff
@@ -272,6 +272,8 @@ func (a *API) addCharmWithAuthorization(args params.AddCharmWithAuth) (params.Ch
 		return params.CharmOriginResult{}, err
 	}
 
+	// TODO(nvinuesa): we must remove this check because we no longer
+	// support charmstore.
```

Comment:
Actually, yes, we can remove this check altogether as it does exactly the same check on this line https://github.com/juju/juju/pull/15226/files#diff-4dcde847939d27dd95cb73d6bb5b7ee55de3c4230e5d36dd308a9f97bc56815bR263 - but uses magic strings :roll_eyes: 

Leave it for now, but the comment is sound.